### PR TITLE
fix mypy errors after the 0.981 release

### PR DESCRIPTION
**Pull request by pmeier**
This fixes the ongoing `mypy` CI failure caused by yesterdays `0.981` release

### CODE REVIEW

File: torchvision/models/_api.py [hunk 1]
```diff
@@ -199,7 +196,7 @@ def list_models(module: Optional[ModuleType] = None) -> List[str]:
     return sorted(models)
 
 
-def get_model_builder(name: str) -> Callable[..., M]:
+def get_model_builder(name: str) -> Callable[..., nn.Module]:
```
Comment (NicolasHug) [inline torchvision/models/_api.py:199]: How of curiosity why do we need to change this one, and not all the other usages of `M` e.g. in `register_model()` above?

File: torchvision/prototype/datasets/utils/_internal.py [hunk 2]
```diff
@@ -72,8 +72,8 @@ def _getattr_closure(obj: Any, *, attrs: Sequence[str]) -> Any:
     return obj
 
 
-def _path_attribute_accessor(path: pathlib.Path, *, name: str) -> D:
-    return cast(D, _getattr_closure(path, attrs=name.split(".")))
+def _path_attribute_accessor(path: pathlib.Path, *, name: str) -> Any:
+    return _getattr_closure(path, attrs=name.split("."))
```
Comment (NicolasHug) [inline torchvision/prototype/datasets/utils/_internal.py:76]: Why is this change needed?
Do you remember why we annotated the return with `D` in the first place?

**Review by NicolasHug — APPROVED** [review]

**Comment by NicolasHug** [comment]
BTW do the changes still pass with older versions of mypy?

File: torchvision/models/_api.py (hunk 1 above)
Comment (pmeier) [reply torchvision/models/_api.py:199]: `TypeVar` is used to create a "link" between parameters. For example I can do something like

```py
def do_nothing(foo: M) -> M:
    return foo
```

This tells `mypy` "whatever we have as input type, we return as output type as well". This is useful in case the function can handle multiple types, e.g.

```py
T = TypeVar("T", int, float)

def add(a: T, b: T) -> T:
    return a + b
``` 

In the case above, there is no input with the type `M` and so `mypy` complains that it can't figure out the `M` in the output. 

Correct me if I'm wrong @datumbox, but I guess the annotation was intended to mean "we return any kind of model" here. Since `M` is defined as 

https://github.com/pytorch/vision/blob/06d9726d66f97ed8e3d3910da246a52be9c6e168/torchvision/models/_api.py#L168

we can simply use `nn.Module` here.

**Comment by pmeier** [comment]
> BTW do the changes still pass with older versions of mypy?

Most likely, but I'll check. The annotations were wrong to begin with, but older versions of `mypy` didn't flag them yet. The new annotations should be correct and thus older versions should also accept them.

File: torchvision/prototype/datasets/utils/_internal.py (hunk 2 above)
Comment (pmeier) [reply torchvision/prototype/datasets/utils/_internal.py:76]: > Why is this change needed?

Same as above. We no longer can use a `TypeVar` as return annotation if we don't have any input with the same type.

> Do you remember why we annotated the return with `D` in the first place?

IIRC, this is a remnant when this function was a local one inside 

https://github.com/pytorch/vision/blob/06d9726d66f97ed8e3d3910da246a52be9c6e168/torchvision/prototype/datasets/utils/_internal.py#L83

In there the annotation was correct and ok, since we have a `D` in the input parameters. At some point we factored them out since `torchdata` doesn't like local functions, and I probably forgot to fix this since `mypy` didn't complain.

**Review by datumbox — APPROVED** [review]
LGTM, thanks!
